pandas/core/generic.py

```diff
@@ -4287,6 +4287,38 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
         else:
             return result
 
+    def _clip_with_one_bound(self, threshold, method, axis, inplace):
+
+        if np.any(isnull(threshold)):
+            raise ValueError("Cannot use an NA value as a clip threshold")
+
+        # method is self.le for upper bound and self.ge for lower bound
+        if is_scalar(threshold) and is_number(threshold):
+            if method.__name__ == 'le':
+                return self._clip_with_scalar(None, threshold, inplace=inplace)
+            else:
+                return self._clip_with_scalar(threshold, None, inplace=inplace)
+
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+
+        subset = method(threshold, axis=axis) | isnull(self)
+
+        # GH #15390
+        if is_scalar(threshold) or is_number(threshold):
+            return self.where(subset, threshold, axis=axis, inplace=inplace)
+
```

Comment:
`clip` will do something like below and using np array or list will cause an error in `where`.  
````
a = pd.DataFrame([[1,2,3], [4,5,6]])
cond = a.ge([1,2,3], axis=1)
a.where(cond, np.asarray([1,2,3]), axis=1)
````

I fell like the name threshold is kind of confusing. It is more like a replacement and got passed as `where`' "other" argument. In the docs, the "other" argument should only take scalar, NDFrame, or callable.

Also, I wrote `threshold = pd.Series(threshold, index=subset.index)` twice. But the reason is kind of different. Should I delete one of them and merge the condition?